app/context/ceramicContext.tsx

```diff
@@ -248,6 +248,35 @@ const startingAllProvidersState: AllProvidersState = Object.values(stampPlatform
   {}
 );
 
+export const handleComposeRetry = async (
+  composeStamps: Stamp[],
+  passDbPassport: Passport,
+  ceramicClient?: ComposeDatabase
+): Promise<SecondaryStorageBulkPatchResponse | void> => {
+  if (ceramicClient) {
+    try {
+      // using stamps as the source of truth, filter stamps for where the issuanceDate does not match the composeStamp if there is a composeStamp with the same provider
+      const stampsToRetry = passDbPassport.stamps.filter((stamp: Stamp) => {
+        const existingStamp = composeStamps.find((composeStamp: Stamp) => stamp.provider === composeStamp.provider);
```

Comment:
this looks like this will only catch stamps that already exist in composedb.  Since stamps don't get updated very often, I imagine the more common form of data discrepancy is going to be a stamp that just doesn't exist at all in composedb, but does in gitcoin's database.  Should we expand this to handle that case as well?

Reply:
I thought the same at first! But the question mark in the next line is doing the heavy lifting

Reply:
It might make things easier to read if we change
`stamp.credential.issuanceDate !== existingStamp?.credential.issuanceDate`
to
`!existingStamp || stamp.credential.issuanceDate !== existingStamp.credential.issuanceDate`

Reply:
Included in commit [d1c7e8b](https://github.com/gitcoinco/passport/pull/2401/commits/d1c7e8b746ed6a0e02edee494ef2067fe8993da0)!